### PR TITLE
[#IC-313]  Update `message-status` model

### PR DESCRIPTION
**Pull request by gquadrati**
#### List of Changes

- Added "isRead" and "isArchived" attributed in MessageStatus model (with `false` as default)
- Updated tests

#### Motivation and Context

Messages pagination and server-side filtering

#### How Has This Been Tested?

Unit tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/models/message_status.ts [hunk 1]
```diff
@@ -22,11 +22,18 @@ export const MESSAGE_STATUS_MODEL_PK_FIELD = "messageId" as const;
...
+  t.interface({
+    messageId: NonEmptyString,
+    status: MessageStatusValue,
+    updatedAt: Timestamp
+  }),
+  t.partial({
+    isRead: t.boolean,
+    // eslint-disable-next-line sort-keys
+    isArchived: t.boolean
+  })
```
Comment (AleDore) [inline src/models/message_status.ts:35]: What if we decide to let these props required with a default ?

Comment (gquadrati) [reply src/models/message_status.ts:35]: You're right, I'll try this way

Comment (michaeldisaro) [reply src/models/message_status.ts:35]: This a good point. By default a message is not read and not archived.

Comment (AleDore) [reply src/models/message_status.ts:35]: We have to check the case that a messageStatus is stored without these props and we perform a retrieve operation that trigger a decode :)

Comment (gquadrati) [reply src/models/message_status.ts:35]: Yes.. I actually don't like very much the idea of storing the two values in the initial steps, when "status" gets changed, but it's more clear when retrieving them

Comment (gquadrati) [reply src/models/message_status.ts]: Done in [0b73fe3](https://github.com/pagopa/io-functions-commons/pull/239/commits/0b73fe318cd16416a6772dd52b2623d480e7812e)
Test updated in [670d905](https://github.com/pagopa/io-functions-commons/pull/239/commits/670d9058d57d0f51598b643d78db13f20319c69a)

File: src/models/message_status.ts [hunk 2]
```diff
@@ -64,12 +64,26 @@ export const getMessageStatusUpdater = (
...
   // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 ): MessageStatusUpdater => status =>
-  messageStatusModel.upsert({
-    kind: "INewMessageStatus",
-    messageId,
-    status,
-    updatedAt: new Date()
-  });
+  pipe(
+    messageStatusModel.findLastVersionByModelId([messageId]),
```
Comment (AleDore) [inline src/models/message_status.ts:68]: Why do we have to retrieve last version before upsert?

Comment (gquadrati) [reply src/models/message_status.ts:68]: Because we cannot upsert "isRead" and "isArchived" without knowing their previous values

Comment (AleDore) [reply src/models/message_status.ts:68]: AFAIK MessageStatusUpdater is called within the ProcessMessage Workflow in order to track if a message is correctly processed, at the same time a citizen cannot trigger a change status (isPending -> true). Btw LGTM to let the updater agnostic related to workflow usage.

Comment (gquadrati) [reply src/models/message_status.ts:68]: Yes, I saw it, but I prefer to be prepared on new statuses and avoid future mistakes

File: src/models/message_status.ts [hunk 3]
```diff
@@ -57,12 +64,26 @@ export const getMessageStatusUpdater = (
...
+    TE.map(
+      O.getOrElse(() => ({
+        messageId,
+        status,
```
Comment (AleDore) [inline src/models/message_status.ts:72]: Maybe `status` should be overridden in any case if a message-status is already present

Comment (gquadrati) [reply src/models/message_status.ts]: ops, my ad. Done in [2f43605](https://github.com/pagopa/io-functions-commons/pull/239/commits/2f43605ca4cbd7ca67cf37324614389ff6a4aa5e)

**Comment by AleDore** [comment]
@gquadrati : it seems integration tests fail

**Comment by gquadrati** [comment]
> @gquadrati : it seems integration tests fail

@AleDore it seems a mailhog test failed. I re-runned integration tests and they worked fine

**Review by AleDore — APPROVED** [review]
LGTM

**Review by michaeldisaro — APPROVED** [review]

**Review by AleDore — APPROVED** [review]
LGTM

**Pull request merged**